src/core/types/elasticsearch/search.ts

```diff
@@ -48,7 +48,7 @@ type ValueTypeOfField<T> = T extends Record<string, string | number>
 
 type MaybeArray<T> = T | T[];
 
-type Fields = Exclude<Required<estypes.SearchRequest>['body']['fields'], undefined>;
+type Fields = MaybeArray<Exclude<Required<estypes.SearchRequest>['body']['fields'], undefined>>;
```

Comment:
Can't we remove a custom type and use `estypes. Fields` instead? It's it became exported recently.

Reply:
I'll give it a shot - previously I think that type was incorrect (e.g. it didn't support arrays or something similar)

Reply:
estypes.Fields is `string | string[]`, but it can also be an array of objects, e.g. `[{ field: '*', include_unmapped: true }]

Reply:
It seems it's already supported?

```ts
SearchRequest.body.fields?: (Field | DateField)[];

export type Field = 'second' | 'minute' | 'hour' | 'day' | 'date' | 'month';
export interface DateField {
  field: Field
  format?: string
  include_unmapped?: boolean
}
```

Reply:
ah yeah, but that's not what is being exported `estypes.Fields`. I can remove the MaybeArray part though, that's not correct anyway. Thanks!